### DISCUSSION TITLE
被adg难住了。

### DISCUSSION THREAD

**Discussion by Lovest20018** [kkkgo/PaoPaoDNS#60] [问答和帮助] [answered]
虚拟机下安装了adg，没跟ppdns同一个ip。
设置adg的上游为ppdns的ip
设置主路由的dhcp下面的dns为adg的ip，
可是却没有任何流量？
是哪里的问题呢？
难道ppdns的转发 端口不是53吗？或者是因为docker的问题？安装adg跟doker在同个虚拟机下。
太小白了，希望得到指导。。。

**Comment by kkkgo** [comment]
没有流量是指什么，我觉得你可以先用ping来排查下你的疑惑。

**Reply by kkkgo** [reply, marked as answer]
参考 https://github.com/kkkgo/PaoPaoDNS/discussions/54#discussioncomment-7289367

**Comment by Lovest20018** [comment]
macvlan网络的话宿主默认和容器是不互通的。哦知道了。怪不得。那我现在除了新建一个其他的虚拟机的话，还有种方式就是用host，然后修改mosdns的端口吗？看到其他问题里不能直接修改，需要映射是吗？可是我用portainer映射貌似没效果。
用命令映射貌似也不行。

**Reply by kkkgo** [reply]
你adg也用macvlan不就得了

**Comment by Lovest20018** [comment]
哦对哦-魔怔了-都忘记了可以另外开个docker了。。非常感谢。。
